charts/gardener/gardenlet/charts/runtime/templates/clusterrole-managed-istio.yaml

```diff
@@ -14,6 +10,10 @@ metadata:
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
 rules:
+{{- if .Values.global.gardenlet.enabled }}
+{{- if .Values.global.gardenlet.config.featureGates }}
+{{- if .Values.global.gardenlet.config.featureGates.ManagedIstio }}
```

Comment:
I'm torn between this solution and simply deleting the role + binding with a check in the `start-gardenlet` script. What I don't like so much is that the role + binding is always deployed even though someone turned off the feature gates right from the beginning. What do you think about checking the feature gate in `start-gardenlet` and deleting at least the binding if ManagedIstio is disabled?

Reply:
This is not applicable only for local setups, also productive landscapes benefit from this change, mostly when the gardenlet is installed in `ManagedSeeds` where the parent gardenlet is rendering and installing the chart and it does take care to remove the orphaned resource of the chart. For example, if gardenlet is deployed once in a Managed Seed with `ManagedIstio=true`, after switching off this feature gate, there is no component to delete the unwanted `gardener.cloud:system:gardenlet:managed-istio` clusterrole(binding) and gardenlet still runs with the ManagedIstio permissions.

Reply:
Okay make sense, thank you. I'm fine with merging it the way it is right now, although I think the clean way of implementing this would be to let the `ManagedSeed` controller delete the role + binding.